core_backend/app/auth/dependencies.py

```diff
@@ -9,62 +9,56 @@
 )
 from jose import JWTError, jwt
 
-from .config import (
-    ACCESS_TOKEN_EXPIRE_MINUTES,
-    JWT_ALGORITHM,
-    JWT_SECRET,
-    USER1_PASSWORD,
-    USER1_RETRIEVAL_KEY,
-    USER1_USERNAME,
-    USER2_PASSWORD,
-    USER2_RETRIEVAL_KEY,
-    USER2_USERNAME,
-)
-from .schemas import AccessLevel, AuthenticatedUser
+from ..database import get_async_session_direct
+from ..users.models import UserDB, get_user_by_token, get_user_by_username
+from ..utils import get_key_hash, setup_logger
+from .config import ACCESS_TOKEN_EXPIRE_MINUTES, JWT_ALGORITHM, JWT_SECRET
+from .schemas import AuthenticatedUser
+
+logger = setup_logger()
 
 bearer = HTTPBearer()
 oauth2_scheme = OAuth2PasswordBearer(tokenUrl="login")
 
 
-USERS = {
-    USER1_USERNAME: {
-        "password": USER1_PASSWORD,
-        "access_level": "fullaccess",
-    },
-    USER2_USERNAME: {
-        "password": USER2_PASSWORD,
-        "access_level": "fullaccess",
-    },
-}
-
-
-def auth_bearer_token(
+async def auth_bearer_token(
     credentials: HTTPAuthorizationCredentials = Depends(bearer),
-) -> str:
+) -> UserDB:
     """
     Authenticate using basic bearer token. Used for calling
     the question-answering endpoints
     """
     token = credentials.credentials
-    # change to "hash check against db" logic later
-    if token not in [USER1_RETRIEVAL_KEY, USER2_RETRIEVAL_KEY]:
-        raise HTTPException(status_code=401, detail="Invalid bearer token")
-    else:
-        return token  # NOT SURE ABOUT THIS - IS THIS REASONABLE?
 
+    asession = await get_async_session_direct()
+    try:
+        user_db = await get_user_by_token(token, asession)
+        return user_db
+    except Exception as err:
+        raise HTTPException(status_code=401, detail="Invalid bearer token") from err
+    finally:
+        await asession.aclose()
```

Comment:
This process of opening a session, doing something, then closing it makes sense here right?

Reply:
My only qualm is that session is not actually closed where it is created. Leads to weird behaviour that is hard to debug. See comment above on one way to do this cleanly.